htdocs/fourn/commande/card.php

```diff
@@ -1100,12 +1106,62 @@
 
 
 	if ($action == 'confirm_delete' && $confirm == 'yes' && $usercandelete) {
-		$result = $object->delete($user);
-		if ($result > 0) {
-			header("Location: ".DOL_URL_ROOT.'/fourn/commande/list.php?restore_lastsearch_values=1');
-			exit;
-		} else {
-			setEventMessages($object->error, $object->errors, 'errors');
+		// Delete existing dispatched lines
+		$errOnDelete = 0;
+		if($stockDelete){
+			$dispatchedLines = $object->getDispachedLines();
+			if (!empty($dispatchedLines)) {
+				foreach ($dispatchedLines as $dispatchedLine) {
+					$db->begin();
```

Comment:
The begin end transaction must not be inside the loop. We must have 1 transaction for all the event so result in database will be everything done or nothing.

Reply:
I modified code to locate correctly begin, rollback and commit instructions for db. 
I noticed that $commande->delete also have a begin, rollback and commit logic on $this->db in the commande.class.php file. 
If two transactions have begun in parallel, I prefered to finish all the logic on $db in the card.php code.